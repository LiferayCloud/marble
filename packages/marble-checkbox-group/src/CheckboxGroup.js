import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './CheckboxGroup.soy.js';

/**
 * CheckboxGroup component.
 */
class CheckboxGroup extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
CheckboxGroup.STATE = {
  /**
   * The list of radio items
   * @type {?Array|undefined}
   * @default undefined
   */
  items: Config.arrayOf(
    Config.shapeOf({
      id: Config.string(),
      checked: Config.bool(),
      label: Config.string(),
      value: Config.string(),
    })
  ),

  /**
   * The name param used on each radio
   * @type {?String}
   * @default undefined
   */
  name: Config.string(),

  /**
   * The style of the radio group
   * @type {!String}
   * @default radio-group
   */
  style: Config.oneOf([
    'checkbox-group',
    'checkbox-group checkbox-group-inline'
  ]).value('checkbox-group'),
};

Soy.register(CheckboxGroup, templates);

export {CheckboxGroup};
export default CheckboxGroup;
