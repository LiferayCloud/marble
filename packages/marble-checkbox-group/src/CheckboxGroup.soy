{namespace CheckboxGroup}

/**
 * This renders the component's whole content.
 */
{template .render}
  {@param? items: list<?>}
  {@param? name: string}
  {@param? style: string}

  <div class="{$style}">
    {if $items}
      {foreach $item in $items}
        {let $checkedAttr kind="attributes"}
          {if $item.checked}
            checked="checked"
          {/if}
        {/let}

        <div class="checkbox">
          <input id="{$item.id}" name="{$name}" value="{$item.value}" type="checkbox" {$checkedAttr}>
          <label for="{$item.id}">{$item.label}</label>
        </div>
      {/foreach}
    {/if}
  </div>
{/template}
