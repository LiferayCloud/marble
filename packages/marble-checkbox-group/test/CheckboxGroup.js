import CheckboxGroup from '../src/CheckboxGroup';

let checkboxGroup;

const items = [
  {
    id: 'checkbox-1-1',
    checked: true,
    label: 'Option 1',
    value: 'Checkbox 1',
  },
  {
    id: 'checkbox-1-2',
    label: 'Option 2',
    value: 'Checkbox 2',
  },
  {
    id: 'checkbox-1-3',
    label: 'Option 3',
    value: 'Checkbox 3',
  },
];

describe('CheckboxGroup', function() {
  afterEach(() => {
    if (checkboxGroup) {
      checkboxGroup.dispose();
    }
  });

  it('should generate the default markup', () => {
    checkboxGroup = new CheckboxGroup({
      name: 'checkbox-group',
      items: items
    });

    expect(checkboxGroup).toMatchSnapshot();
  });

  it('should render with inline style', () => {
    checkboxGroup = new CheckboxGroup({
      name: 'checkbox-group',
      style: 'checkbox-group checkbox-group-inline',
      items: items
    });

    expect(checkboxGroup).toMatchSnapshot();
  });
});
