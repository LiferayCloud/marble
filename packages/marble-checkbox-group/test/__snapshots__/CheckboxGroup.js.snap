// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup should generate the default markup 1`] = `
<div class="checkbox-group">
  <div class="checkbox">
    <input id="checkbox-1-1" name="checkbox-group" value="Checkbox 1" type="checkbox" checked="">
    <label for="checkbox-1-1">Option 1</label>
  </div>
  <div class="checkbox">
    <input id="checkbox-1-2" name="checkbox-group" value="Checkbox 2" type="checkbox">
    <label for="checkbox-1-2">Option 2</label>
  </div>
  <div class="checkbox">
    <input id="checkbox-1-3" name="checkbox-group" value="Checkbox 3" type="checkbox">
    <label for="checkbox-1-3">Option 3</label>
  </div>
</div>
`;

exports[`CheckboxGroup should render with inline style 1`] = `
<div class="checkbox-group checkbox-group-inline">
  <div class="checkbox">
    <input id="checkbox-1-1" name="checkbox-group" value="Checkbox 1" type="checkbox" checked="">
    <label for="checkbox-1-1">Option 1</label>
  </div>
  <div class="checkbox">
    <input id="checkbox-1-2" name="checkbox-group" value="Checkbox 2" type="checkbox">
    <label for="checkbox-1-2">Option 2</label>
  </div>
  <div class="checkbox">
    <input id="checkbox-1-3" name="checkbox-group" value="Checkbox 3" type="checkbox">
    <label for="checkbox-1-3">Option 3</label>
  </div>
</div>
`;
